man/org.freedesktop.login1.xml

```diff
@@ -992,6 +992,14 @@ node /org/freedesktop/login1/user/_1000 {
     methods:
       Terminate();
       Kill(in  i signal_number);
+      SecureLock();
+      DelaySecureLock(out h delay_fd);
+      InhibitAutoSecureLock(out h inhibitor_fd);
```

Comment:
hmm, can't we make this part of the usual inhibitor logic?

Reply:
ah, i guess you want this per-user while the existing inhibitor logic is per-system.

so i'd really like to keep the door open to expose a screen lock in similar ways one day. hence I think this should take a bit mask, and one bit should be for the secure lock thing.

(the per-system inhibitor locks take this weird ":" separated string. it should really be a bitmask though. The ":" thing was a historical mistake we should not repeat).

Reply:
Right, this is per-user but normal inhibitors are system-wide.

I can't reuse this for normal screen locks either, because screen locks are per-session, which is again not per-user.

If you want a more flexible inhibitors API anyway, I have a few questions:
- do you want the "who"/"why" fields?
- will these need to be listed in any way?
- there are some invalid configurations, like delaying auto-secure-lock. I would also like to prevent callers from blocking secure lock in general (letting clients opt the user out of a security feature even if the user manually triggers it seems like a bad idea...). What to do about them?

Reply:
and better to prohibit too much initially (and losen things later maybe) than the reverse.

Reply:
I decided to just let callers fully inhibit secure locking. Maybe there's a use-case. I do prevent callers from delaying things that don't make sense to delay, however.